### PR TITLE
fix: decimals

### PR DESCRIPTION
**Pull request by trungtt198x**
# Description of change

Adapt the current code to use 18 decimals for SMR native on L2

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes
- [ ] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Comment by trungtt198x** [comment]
@begonaalvarezd 
As discussed on Slack, pls help review/merge this PR.
Thanks

File: src/lib/constants.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
 export const Bech32AddressLength = 63;
 export const EVMAddressLength = 42;
 export const NativeTokenIDLength = 38;
+
+export const BASE_TOKEN_DECIMALS = 18;
```
Comment (begonaalvarezd) [inline src/lib/constants.ts:5]: I would rename this to clarify is the L2 base token

File: src/lib/iscmagic/iscmagic.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export class ISCMagic {
 
   public async getBaseTokens(eth: Eth, account: string): Promise<number> {
     const addressBalance = await eth.getBalance(account);
-    const balance = BigInt(addressBalance) / BigInt(1e12);
+    const balance = BigInt(addressBalance); /*/ BigInt(1e12)*/;
```
Comment (begonaalvarezd) [inline src/lib/iscmagic/iscmagic.ts:40]: Could we just remove the comment?

File: src/lib/constants.ts (hunk 1 above)
Comment (begonaalvarezd) [reply src/lib/constants.ts:5]: We could also make use of these constants in other places like
- https://github.com/trungtt198x/evm-toolkit/blob/ee02adfc169e114d3426baf7d53ddf0bb87e00c1/src/lib/withdraw/actions/add-network-to-metamask.ts#L21-L22
- https://github.com/trungtt198x/evm-toolkit/blob/ee02adfc169e114d3426baf7d53ddf0bb87e00c1/src/lib/withdraw/actions/add-token-to-metamask.ts#L20
- https://github.com/trungtt198x/evm-toolkit/blob/ee02adfc169e114d3426baf7d53ddf0bb87e00c1/src/routes/sections/wrap-section.svelte#L201
wdyt?

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
Added some minor comments. 
Additionally, withdraw doesnt work for me anymore 🙏🏼 
image

File: src/lib/constants.ts (hunk 1 above)
Comment (trungtt198x) [reply src/lib/constants.ts:5]: Agree

File: src/lib/iscmagic/iscmagic.ts (hunk 2 above)
Comment (trungtt198x) [reply src/lib/iscmagic/iscmagic.ts:40]: sure

**Review by begonaalvarezd — APPROVED** [review]
🚀 works

**Pull request merged**